File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/Region.ts
import { Country } from './Country';

export class Region {
    //id: number;
  id: string;
    name: string;
  country: Country;
  constructor(options: {
    //id: number,
    id: string,
    name: string,
    country: Country
  }) {
    if (options) {
      this.id = options.id;
      this.name = options.name;
      this.country = new Country(options.country);
    }
  }
}

File: SiteVisitsNetCore2020Web/Models/Region.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SiteVisitsNetCore2020Web.Models
{
    public class Region
    {
        //public int Id { get; set; }
        public Guid Id { get; set; }
        public string Name { get; set; }
        public virtual Country Country { get; set; }
        public DateTime FirstOccurrence { get; set; }
        public DateTime DateCreated { get; set; }

    }
}
